### ISSUE TITLE
Catalog v2 includes master branch in clone by default

### ISSUE THREAD

**Issue by brettmorien** [rancher/rancher#30596]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Use terraform to create a rancher2_catalog_v2 resource and then a rancher2_app_v2 using that catalog. Set the catalog git_repo to a Github repository with a 'main' branch as the default with no 'master' branch.

```
resource "rancher2_catalog_v2" "indeed_harbor_webhook" {
  name       = "indeed-harbor-webhook"
  git_repo   = "https://github.com/indeedeng-alpha/harbor-container-webhook"
  git_branch = "main"
  cluster_id = rancher2_cluster_sync.this.cluster_id
}

resource "rancher2_app_v2" "harbor_container_webhook" {
  repo_name     = rancher2_catalog_v2.indeed_harbor_webhook.name
  chart_name    = "harbor-container-webhook"
  chart_version = "0.2.0"
...
}
```

**Result:**
Intermittent failure to install the app because the 'master' branch does not exist.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.2
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Imported
- Machine type (cloud/VM/metal) and specifications (CPU/memory): m5.2xlarge
- Kubernetes version (use `kubectl version`): 1.18 EKS

Terraform error
```
Error: failed to get chart info harbor-container-webhook:0.2.0 from catalog v2 indeed-harbor-webhook: Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [code=ServerError, message=failed to find chartName harbor-container-webhook version 0.2.0: NotFound 404] from [https://rancher.stellar.eadp.cloud/k8s/clusters/c-n54gz/v1/catalog.cattle.io.clusterrepos/indeed-harbor-webhook?link=info&chartName=harbor-container-webhook&version=0.2.0]

on stellar/harbor.tf line 153, in resource "rancher2_app_v2" "harbor_container_webhook":

153: resource "rancher2_app_v2" "harbor_container_webhook" {
...
```

From the API endpoint:
```
spec:
  enabled: true
  gitBranch: main
  gitRepo: https://github.com/brettmorien/harbor-container-webhook
status:
  branch: main
  commit: 1b0db64c1be6e39d668101b8cca5a383852a3c96
  conditions:
  - lastUpdateTime: "2020-12-22T22:02:06Z"
    message: |
      git clone --depth=1 -n --branch master https://github.com/brettmorien/harbor-container-webhook management-state/git-repo/indeed-harbor-webhook/b9b60aadd47c24859dc061474a9f30da4b7d5fadd68fd5903812b29ea2a09961 error: exit status 128, detail: Cloning into 'management-state/git-repo/indeed-harbor-webhook/b9b60aadd47c24859dc061474a9f30da4b7d5fadd68fd5903812b29ea2a09961'...
      warning: Could not find remote branch master to clone.
      fatal: Remote branch master not found in upstream origin
    reason: Error
    status: "False"
    type: FollowerDownloaded
```

Editing and saving the catalog from the Rancher UI with the same value (main branch) fixes the issue.

**Comment by brettmorien** [comment]
Additional information. A member of our team tracked it to the internals of the Wrangler library used for git cloning. On this line: https://github.com/rancher/wrangler/blob/master/pkg/git/git.go#L130 it explicitly uses the "master" branch.
```
func (g *Git) Ensure(commit string) error {
	if err := g.clone("master"); err != nil {
		return err
	}
```

I will file a bug over in that library, but this should be accounted for here as well.

**Comment by brettmorien** [comment]
Update: The fix for the issue in Wrangler has been accepted and merged. Fixing this issue in Rancher just requires a release to Wrangler and an update to take that new version here. Fixing this would be good to further deprecation of the problematic 'master' terminology and support those projects that have already taken that step.

**Comment by brettmorien** [comment]
Update: This fix in Wrangler has been released in Wrangler 0.8.0. Please take this upgrade when possible.
https://github.com/rancher/wrangler/releases/tag/v0.8.0

**Comment by peter-accedian** [comment]
I can confirm Rancher 2.5.7 still suffers from this bug when operated with terraform (the UI seems to be reading and deploying charts ok from set branch), setting catalog branch to any other branch other than master results in error 500 chart not found.

fastest workaround is to move your catalog to master branch. 

i tried to find Wrangler in the list of containers that RKE generates into initial config file and its not there, to be honest i'm not sure where would I need to to change as I'm not running Wrangler (i don't think)

can anyone confirm fix is coming?
